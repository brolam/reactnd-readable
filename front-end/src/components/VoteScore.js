import React from 'react'
import PropTypes from 'prop-types'

function VoteScore(props){
    return([
      <button key="buttonLiked" className="liked" href="/"  >Liked <span>{props.voteScore}</span></button>,
      <button key="buttonNotLiked" className="not-liked" href="/" >Not like</button>
    ])
} 

VoteScore.propTypes = {
  entityType: PropTypes.string.isRequired,
  entityId: PropTypes.string.isRequired,
  voteScore: PropTypes.number.isRequired
}

export default VoteScore