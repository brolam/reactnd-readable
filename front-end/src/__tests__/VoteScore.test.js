import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import VoteScore from '../components/VoteScore'

test('render without error', () => {
  ReactDOM.render(
    <VoteScore
      entityType="post"
      entityId="8xf0y6ziyjabvozdd253nd"
      voteScore={5}
    />, document.createElement('div'));
})

test('last Snapshot', () => {
  const voteScore = renderer.create(
    <VoteScore
      entityType="post"
      entityId="8xf0y6ziyjabvozdd253nd"
      voteScore={5}
    />
  );
  let tree = voteScore.toJSON();
  expect(tree).toMatchSnapshot();
})