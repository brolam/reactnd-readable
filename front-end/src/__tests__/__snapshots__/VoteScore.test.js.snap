// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`last Snapshot 1`] = `
Array [
  <button
    className="liked"
    href="/"
>
    Liked 
    <span>
        5
    </span>
</button>,
  <button
    className="not-liked"
    href="/"
>
    Not like
</button>,
]
`;
